src/main/java/com/bigchaindb/model/Input.java

```diff
@@ -40,14 +43,32 @@ public String getFullFillment() {
 	}
 
 	/**
-	 * Sets the full fillment.
+	 * Gets the fulfillment details.
+	 *
+	 * @return fulfillment details.
+	 */
+	public Details getFullFillmentDetails() {
+		return fullFillmentDetails;
+	}
+
+	/**
+	 * Sets the full fillment as a string.
 	 *
 	 * @param fullFillment the new full fillment
 	 */
 	public void setFullFillment(String fullFillment) {
 		this.fullFillment = fullFillment;
 	}
 
+	/**
+	 * Sets the full fillment as a set of details.
+	 *
+	 * @param fullFillment the new full fillment
+	 */
+	public void setFullFillment(Details fullFillment) {
+		this.fullFillmentDetails = fullFillment;
```

Comment:
overlaod of `setFullFillment` to allow passing `Details` directly in. the `InputSerializer` later picks it up, if it is set to anything other than `null`, and uses it in the request according to bigchain's YAML schema files.

Reply:
accepted